packaging/hudi-flink-bundle/pom.xml

```diff
@@ -639,7 +639,6 @@
       <groupId>org.apache.orc</groupId>
       <artifactId>orc-core</artifactId>
       <version>${orc.version}</version>
-      <classifier>nohive</classifier>
```

Comment:
looks like flink bundle does not shade orc. this change may impact on flink usage. let's have this change and shading in a separate PR?

Reply:
We can't have it in a separate PR, since our code-change is making it incompatible w/ previous "nohive" version.

And we can't shade Orc or Parquet or any other format which is interfacing w/ the execution engine b/c we share the classes w/ it (meaning that we pass these classes' objects from Hudi to the engine, and back)

Reply:
in principle hudi should not include file formats in the bundles, and just use provided ones and also make sure hudi using APIs correctly. But if we include some libs, we should shade for hudi's use. There are inconsistencies about this across bundles, for flink bundle, avro is shaded but parquet is not. When you say "interfacing w/ the execution engine" it's meant for class loading, which should be addressed at code level, like avoid calling APIs that load hardcoded classes, which i think it's rare situations (Kryo is one of those)

Reply:
@xushiyan let me take back my 2d point: 

We should actually not be depending on this modules directly -- for reading these file-formats we're relying on readers provided by the execution engine(s) and therefore should not depend on these modules directly for that. 

The only other use-case i see us depending on these modules for are schema conversions in hive-sync, hadoop-mr bundles. For that we can (and should) shade these deps indeed.

Reply:
> When you say "interfacing w/ the execution engine" it's meant for class loading, which should be addressed at code level, like avoid calling APIs that load hardcoded classes, which i think it's rare situations (Kryo is one of those)

It's not only that, it's that you might be creating an object of class X and passing it to the execution engine -- in that case both have to share the class X (therefore you can't shade it in one of them). 

Please check my other comment though, i don't think this is the case for us, and so we should be able to shade actually.

Reply:
understood. we're aligned then. also the orc version we use with flink bundle should align with flink's orc versions under flink profiles. we may have to use a separate version property like `orc.flink.version` vs `orc.spark.version` in case there are discrepancies

Reply:
@danny0405 can you help us understand how Orc is used w/in the Flink bundle?